pkg/namespace/namespace.go

```diff
@@ -0,0 +1,76 @@
+// package namespace contains required code to interact with a Cloud Platform namespace.
+package namespace
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/google/go-github/v38/github"
+	"golang.org/x/oauth2"
+)
+
+// ChangedInPR takes a GitHub branch reference (usually provided by a GitHub Action), a
+// personal access token with Read org permissions, the name of a repository and the owner.
+// It queries the GitHub API for all changes made in a PR. If the PR contains changes to a namespace
+// it returns a deduplicated slice of namespace names.
+func ChangedInPR(branchRef, token, repo, owner string) ([]string, error) {
+	if token == "" {
+		return nil, errors.New("You must have a valid GitHub token.")
+	}
+
+	client := githubClient(token)
+
+	str := strings.Split(branchRef, "/")
```

Comment:
Can you please add an example comment on how branchRef would look like so it is easy to understand the fornat

Reply:
good thinking. Adding now

Reply:
a7c5395